LoRaEngine/modules/LoRaWanNetworkSrvModule/Logger/LoRaConsoleLogger.cs

```diff
@@ -0,0 +1,166 @@
+    using System.Collections.Generic;
+
+    public class LoRaConsoleLoggerConfiguration
+    {
+        public LogLevel LogLevel { get; set; }
+        public EventId EventId { get; set; }
+        public bool UseScopes { get; set; } = true;
+    }
+
+    public class LoRaConsoleLoggerProvider : ILoggerProvider
+    {
+        private readonly ConcurrentDictionary<string, LoRaConsoleLogger> loggers = new();
+        private readonly IDisposable onChangeToken;
+
+        public LoRaConsoleLoggerProvider(IOptionsMonitor<LoRaConsoleLoggerConfiguration> config)
+        {
+            _ = config ?? throw new ArgumentNullException(nameof(config));
+
+            this.onChangeToken = config.OnChange(UpdateConfiguration);
+            UpdateConfiguration(config.CurrentValue);
+        }
+
+        private void UpdateConfiguration(LoRaConsoleLoggerConfiguration configuration)
+        {
+            Configuration = configuration;
+            ScopeProvider = Configuration.UseScopes ? new LoggerExternalScopeProvider() : null;
+        }
+
+        public LogLevel LogLevel => Configuration.LogLevel;
+        public EventId EventId => Configuration.EventId;
+        internal LoRaConsoleLoggerConfiguration Configuration { get; private set; }
+        internal IExternalScopeProvider ScopeProvider { get; private set; }
+
+        public ILogger CreateLogger(string categoryName) =>
+            this.loggers.GetOrAdd(categoryName, name => new LoRaConsoleLogger(this));
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                this.loggers.Clear();
+                this.onChangeToken.Dispose();
+            }
+        }
+    }
+
+    /// <summary>
+    /// The LoRaConsoleLogger is a simple console logger that is excluding category and timestamp information.
+    /// It's a plain message output with support for structured parameters. Scopes are supported and can
+    /// be used to include standard output values in the message.
+    /// </summary>
+    public class LoRaConsoleLogger : ILogger
+    {
+        private readonly LoRaConsoleLoggerProvider provider;
+
+        public LoRaConsoleLogger(LoRaConsoleLoggerProvider consoleLoggerProvider)
+        {
+            this.provider = consoleLoggerProvider;
+        }
+
+        public IDisposable BeginScope<TState>(TState state) =>
+            this.provider.ScopeProvider is { } scopeProvider ? scopeProvider.Push(state) : default;
+
+        public bool IsEnabled(LogLevel logLevel) => logLevel >= this.provider.LogLevel;
+
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            if (!IsEnabled(logLevel)) return;
+            _ = formatter ?? throw new ArgumentNullException(nameof(formatter));
```

Comment:
Argument checks should come first:

```suggestion
            _ = formatter ?? throw new ArgumentNullException(nameof(formatter));

            if (!IsEnabled(logLevel))
                return;
```

Reply:
As discussed: this was pure perf motivated. However, we can use the DebugLogger?.Log design to overcome putting things on the stack and then this becomes a non-issue.